includes/woocommerce-memberships/class-admin.php

```diff
@@ -0,0 +1,149 @@
+	 *
+	 * @var string
+	 */
+	const NETWORK_MANAGED_META_KEY = '_managed_by_newspack_network';
+
+	/**
+	 * Initializer.
+	 */
+	public static function init() {
+		add_action( 'add_meta_boxes', array( __CLASS__, 'add_meta_box' ), 100 ); // Memberships plugin does something there and we need a higher priority.
+		add_action( 'save_post', array( __CLASS__, 'save_meta_box' ) );
+		add_filter( 'get_edit_post_link', array( __CLASS__, 'get_edit_post_link' ), 10, 2 );
+		add_filter( 'post_row_actions', array( __CLASS__, 'post_row_actions' ), 99, 2 ); // After the Memberships plugin.
+	}
+
+	/**
+	 * Adds a meta box to the membership plan edit screen.
+	 */
+	public static function add_meta_box() {
+		add_meta_box(
+			'newspack-network-memberships-meta-box',
+			__( 'Newspack Network', 'newspack-network' ),
+			array( __CLASS__, 'render_meta_box' ),
+			self::MEMBERSHIPS_CPT,
+			'advanced'
+		);
+	}
+
+	/**
+	 * Renders the meta box.
+	 *
+	 * @param \WP_Post $post The post object.
+	 */
+	public static function render_meta_box( $post ) {
+		$network_id = get_post_meta( $post->ID, self::NETWORK_ID_META_KEY, true );
+
+		wp_nonce_field( 'newspack_network_save_membership_plan', 'newspack_network_save_membership_plan_nonce' );
+		?>
+		<label for="newspack-network-id"><?php esc_html_e( 'Network ID', 'newspack-network' ); ?></label>
+		<input type="text" id="newspack-network-id" name="newspack_network_id" value="<?php echo esc_attr( $network_id ); ?>" />
+		<p><?php esc_html_e( 'If a Network ID is informed, the user memberships will be propagated to the network. The user will be granted with the membership with the matching Network ID in all other sites in the network.', 'newspack-network' ); ?></p>
+		<?php
+	}
+
+	/**
+	 * Saves the meta box.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public static function save_meta_box( $post_id ) {
+
+		$post_type = sanitize_text_field( $_POST['post_type'] ?? '' );
+
+		if ( self::MEMBERSHIPS_CPT !== $post_type ) {
+			return;
+		}
+
+		if ( ! isset( $_POST['newspack_network_save_membership_plan_nonce'] ) ||
+			! wp_verify_nonce( sanitize_text_field( $_POST['newspack_network_save_membership_plan_nonce'] ), 'newspack_network_save_membership_plan' )
+		) {
+			return;
+		}
+
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+
+		$post_type_object = get_post_type_object( $post_type );
+
+		if ( ! current_user_can( $post_type_object->cap->edit_post, $post_id ) ) {
+			return;
+		}
+
+		$network_id = sanitize_text_field( wp_unslash( $_POST['newspack_network_id'] ?? '' ) );
+		update_post_meta( $post_id, self::NETWORK_ID_META_KEY, $network_id );
+	}
+
+	/**
+	 * Filters the edit post link for user memberships managed in another site of the network.
+	 *
+	 * @param string $link The edit post link.
+	 * @param int    $post_id The post ID.
+	 * @return string
+	 */
+	public static function get_edit_post_link( $link, $post_id ) {
```

Comment:
The membership on the "receiving" site is still editable if visited directly (manually typing the post edit link). The post-save redirect goes to the wrong link, though (see comment on `post_row_actions` method). The membership updated this way (from the receiving site) is not updated on the origin site. This might be confusing. 

If the update is supposed to be one-way, the membership edit view should not be editable, just like post edit view of a distributed post.